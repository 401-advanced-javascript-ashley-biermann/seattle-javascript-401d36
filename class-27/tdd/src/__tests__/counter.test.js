import React from 'react';
// shallow lets us look at just the JSX that is returns, mount allows us to check state / prop stuff
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Counter from '../components/counter.js';


// counter should maintain a state property called count, which gets increments on button click
// counter should display the count to the user

describe('Counter tests', () => {
  it('should render at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('p').exists()).toBe(true);
    expect(app.find('button').exists()).toBe(true);
  });

  it('should update state on click', () => {
    let app = mount(<Counter />);
    let increment = app.find('#increment');
    expect(increment.exists()).toBeTruthy();
    expect(app.state('count')).toBe(0);
    increment.simulate('click');
    expect(app.state('count')).toBe(1);
  });

  it('renders correctly', () => {
    const DOM = renderer.create(<Counter />).toJSON();
    expect(DOM).toMatchSnapshot();
  });
});