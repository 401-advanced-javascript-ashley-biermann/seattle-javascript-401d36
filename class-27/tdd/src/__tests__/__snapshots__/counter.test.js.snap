// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter tests renders correctly 1`] = `
<div>
  <form>
    <button
      id="increment"
      onClick={[Function]}
    >
      Add +1
    </button>
  </form>
  <p>
    0
    ;
  </p>
  <button
    onClick={[Function]}
  >
    Clear
  </button>
</div>
`;
